### PR TITLE
adding code for warehouseStateFact gold

### LINKED ISSUE #1252: Create Gold module for Warehouse State Fact
Add a new module in Gold layer for warehouse_state_fact

### PR DESCRIPTION
**Pull request by aman-db**
closes #1252 

Added module for warehouse state fact gold table and view
added another static table for warehouse dbu calculation

### CODE REVIEW

File: src/main/scala/com/databricks/labs/overwatch/pipeline/GoldTransforms.scala [hunk 1]
```diff
@@ -950,6 +951,113 @@ trait GoldTransforms extends SparkSessionWrapper {
...
+      ("AZURE","2X-Small","E8ds v4",1,4),
+      ("AZURE","X-Small","E8ds v4",2,6),
+      ("AZURE","Small","E16ds v4",4,12),
+      ("AZURE","Medium","E32ds v4",8,24),
+      ("AZURE","Large","E32ds v4",16,40),
+      ("AZURE","X-Large","E64ds v4",32,80),
+      ("AZURE","2X-Large","E64ds v4",64,144),
+      ("AZURE","3X-Large","E64ds v4",128,272),
+      ("AZURE","4X-Large","E64ds v4",256,528)
+    )
```
Comment (neilbest-db) [inline src/main/scala/com/databricks/labs/overwatch/pipeline/GoldTransforms.scala:978]: It looks like you have duplicated the data that is also in `src/main/resources/Warehouse_DBU_Details.csv`, correct?  why not load it from the resource file in the JAR (assuming that the resource files are included in the assembly JAR)?  `scala.io.Source.fromResource` from the standard library should be the natural way to do this.

**Review by neilbest-db — CHANGES_REQUESTED** [review]
Please consider the comment I made re: duplicate data in `GoldTransforms.scala`.

File: src/main/scala/com/databricks/labs/overwatch/pipeline/GoldTransforms.scala (hunk 1 above)
Comment (aman-db) [reply src/main/scala/com/databricks/labs/overwatch/pipeline/GoldTransforms.scala]: @neilbest-db this is done pls check

**Review by neilbest-db — APPROVED** [review]

**Comment by neilbest-db** [comment]
@aman-db , I see that you [force-pushed](https://github.com/databrickslabs/overwatch/compare/527996ccb51a3adfb9da18e4f8e3fc10fa67a6ba..67f64d2fe27e95724ddf3501411d0daee68852d8) something here yesterday.  Is the next step to edit the target branch of the PR or what?
image

**Comment by gueniai** [comment]
@aman-db reminder to resolve conflicts on this one

**Pull request merged**